src/test/java/org/openhab/tools/analysis/checkstyle/test/MarkDownCheckTest.java

```diff
@@ -0,0 +1,247 @@
+ *
+ * @author Erdoan Hadzhiyusein - Initial implementation
+ */
+public class MarkDownCheckTest extends AbstractStaticCheckTest {
+    private static final String README_MD_CHECK_TEST_DIRECTORY_NAME = "markDownCheckTest";
+    private static final String ADDED_README_IN_BUILD_PROPERTIES_MSG = "README.MD file must not be added to the bin.includes property";
+    private static final String ADDED_DOC_FOLDER_IN_BUILD_PROPERTIES_MSG = "The doc folder must not be added to the bin.includes property";
+    private DefaultConfiguration config;
+
+    @Before
+    public void setUpClass() {
+        createValidConfig();
+    }
+
+    @Test
+    public void testHeader() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(1, "Missing an empty line after the Markdown header (#).");
+        verifyMarkDownFile("testHeader", expectedMessages);
+    }
+
+    @Test
+    public void testDisabledNodeVisit() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testDisabledNodeVisit", expectedMessages);
+    }
+
+    @Test
+    public void headerAtEndOfFile() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(6, "There is a header at the end of the Markdown file. Please consider adding some content below.");
+        verifyMarkDownFile("testHeaderAtEndOfFile", expectedMessages);
+    }
+
+    @Test
+    public void testSpecialHeader() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testHeaderOutOfBounds", expectedMessages);
+    }
+
+    @Test
+    public void testEmptyLineBeforeList() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(1, "The line before a Markdown list must be empty.");
+        verifyMarkDownFile("testEmptyLineBeforeList", expectedMessages);
+    }
+
+    @Test
+    public void testEmptyLineAfterList() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(7, "The line after a Markdown list must be empty.");
+        verifyMarkDownFile("testEmptyLineAfterList", expectedMessages);
+    }
+
+    @Test
+    public void testListAtEndOfFile() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testListAtEndOfFile", expectedMessages);
+    }
+
+    @Test
+    public void testCodeFormattedListBlock() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testCodeFormattedListBlock", expectedMessages);
+    }
+
+    @Test
+    public void testEmptyLinedCodeBlock() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testEmptyLinedCodeBlock", expectedMessages);
+    }
+
+    @Test
+    public void testCodeFormattedListElement() throws Exception {
+        String[] expectedMessages = EMPTY_STRING_ARRAY;
+        verifyMarkDownFile("testCodeFormattedListElement", expectedMessages);
```

Comment:
You could consider adding a separate method to handle the case where no error occurred no you repeat `String[] expectedMessages = EMPTY_STRING_ARRAY;` every time